test/test.js

```diff
@@ -19423,6 +19425,386 @@
+      assert.deepEqual(_.toNumber(' -0o12345 '), NaN);
+      assert.strictEqual(_.toNumber(' 0O12345 '), 5349);
+      assert.deepEqual(_.toNumber(' +0O12345 '), NaN);
+      assert.deepEqual(_.toNumber(' -0O12345 '), NaN);
+      assert.strictEqual(_.toNumber(' 0x1a2b3c'), 1715004);
+      assert.deepEqual(_.toNumber(' +0x1a2b3c '), NaN);
+      assert.deepEqual(_.toNumber(' -0x1a2b3c '), NaN);
+      assert.strictEqual(_.toNumber(' 0X1A2B3C '), 1715004);
+      assert.deepEqual(_.toNumber(' +0X1A2B3C '), NaN);
+      assert.deepEqual(_.toNumber(' -0X1A2B3C '), NaN);
+      assert.deepEqual(_.toNumber(' 0b1010102 '), NaN);
+      assert.deepEqual(_.toNumber(' +0b1010102 '), NaN);
+      assert.deepEqual(_.toNumber(' -0b1010102 '), NaN);
+      assert.deepEqual(_.toNumber(' 0B1010102 '), NaN);
+      assert.deepEqual(_.toNumber(' +0B1010102 '), NaN);
+      assert.deepEqual(_.toNumber(' -0B1010102 '), NaN);
+      assert.deepEqual(_.toNumber(' 0o123458 '), NaN);
+      assert.deepEqual(_.toNumber(' +0o123458 '), NaN);
+      assert.deepEqual(_.toNumber(' -0o123458 '), NaN);
+      assert.deepEqual(_.toNumber(' 0O123458 '), NaN);
+      assert.deepEqual(_.toNumber(' +0O123458 '), NaN);
+      assert.deepEqual(_.toNumber(' -0O123458 '), NaN);
+      assert.deepEqual(_.toNumber(' 0x1a2b3x '), NaN);
+      assert.deepEqual(_.toNumber(' +0x1a2b3x '), NaN);
+      assert.deepEqual(_.toNumber(' -0x1a2b3x '), NaN);
+      assert.deepEqual(_.toNumber(' 0X1A2B3X '), NaN);
+      assert.deepEqual(_.toNumber(' +0X1A2B3X '), NaN);
+      assert.deepEqual(_.toNumber(' -0X1A2B3X '), NaN);
+      assert.deepEqual(_.toNumber(' -0b '), NaN);
+      assert.deepEqual(_.toNumber(' -0o '), NaN);
+      assert.deepEqual(_.toNumber(' -0x '), NaN);
+    });
+
+    QUnit.test('should convert boolean literals and objects', function(assert) {
+      assert.expect(6);
+
+      assert.strictEqual(1 / _.toNumber(false), INFINITY);
+      assert.strictEqual(_.toNumber(true), 1);
+      assert.strictEqual(1 / _.toNumber(new Boolean(false)), INFINITY);
+      assert.strictEqual(_.toNumber(new Boolean(true)), 1);
+      assert.deepEqual(_.toNumber('false'), NaN);
+      assert.deepEqual(_.toNumber('true'), NaN);
+    });
+
+    QUnit.test('should convert dates', function(assert) {
+      assert.expect(2);
+
+      var now = new Date();
+      assert.strictEqual(_.toNumber(now), now.getTime());
+      assert.deepEqual(_.toNumber(new Date(MAX_INTEGER)), NaN);
+    });
+
+    QUnit.test('should convert RegExp literals and objects', function(assert) {
+      assert.expect(2);
+
+      assert.deepEqual(_.toNumber(/abc/i), NaN);
+      assert.deepEqual(_.toNumber(new RegExp('abc', 'i')), NaN);
+    })
+
+    QUnit.test('other objects', function(assert) {
+      assert.expect(14);
+
+      assert.deepEqual(_.toNumber({}), NaN);
+      assert.deepEqual(_.toNumber({
+        valueOf: '1.1'
+      }), NaN);
+      assert.strictEqual(_.toNumber({
```

Comment:
I should be able to roll it into something much smaller once I make it robust for the environments that you are testing on Travis.